lib/seahorn/BvOpSem2FatMemMgr.cc

```diff
@@ -323,96 +389,109 @@ class FatMemManager : public OpSemMemManager {
   FatMemValTy MemSet(PtrTy ptr, Expr _val, unsigned len, FatMemValTy mem,
                      uint32_t align) override {
     return mkFatMem(
-        m_mem.MemSet(mkRawPtr(ptr), _val, len, mkRawMem(mem), align));
+        m_main.MemSet(mkRawPtr(ptr), _val, len, mkRawMem(mem), align),
+        m_main.MemSet(mkRawPtr(ptr), _val, len, mkSlot0Mem(mem), align),
```

Comment:
Multiple problems:
1. `m_main` is not the right manager for slot0 and slot1
2. `_val` is not the right value to be written to the slot

Same as in `storeIntToMem`. Might be easier to leave unused slots uninitialized.

Reply:
For 1, you may be looking at an outdated version. I have already fixed it as follows

 ```
   return mkFatMem(
        m_main.MemSet(mkRawPtr(ptr), _val, len, mkRawMem(mem), align),
        m_slot0.MemSet(mkRawPtr(ptr), m_ctx.alu().si(0, g_slotBitWidth), len,
                       mkSlot0Mem(mem), align),
        m_slot1.MemSet(mkRawPtr(ptr), m_ctx.alu().si(0, g_slotBitWidth), len,
                       mkSlot1Mem(mem), align));
```
2. I don't think`_val` is wrong since slot values are don't care?
3. Ack on not initializing.

Reply:
yes, old code. the new is good. But would forgo initialization right now.